src/attr/_make.py

```diff
@@ -1244,12 +1248,18 @@ def __repr__(self):
         try:
             result = [class_name, "("]
             first = True
-            for name in attr_names:
+            for attr in attrs:
+                if not attr.repr:
```

Comment:
You are going to filter out falsy-callables here "accidentally".

Reply:
Yes, please be explicit here.

Reply:
refactored it a bit in 23515814a717ad83869dba252b401582a5f7274b

strict `bool` checking, and only do the filtering and ‘which repr to use?’ detection once, instead of per `repr()` call.